src/utils/compute-title.js

```diff
@@ -0,0 +1,54 @@
+const MAX_TITLE_LENGTH = 106;
+const SEPARATOR = ' | ';
+
+const computePath = (parents, maxAllowedPathSize) => {
+    let testedPath = '';
+    let path = '';
+
+    for (let i = 0; i < parents.length; i++) {
+        testedPath = '/' + parents[i] + testedPath;
+        if (testedPath.length > maxAllowedPathSize) {
+            return '...' + path;
+        }
+
+        path = testedPath;
+    }
+
+    return path;
+};
+
+const computePageTitleWithFirstDirectory = (pageTitle, parents) => {
+    return pageTitle + (parents.length > 1 ? '...' : '') + '/' + parents[0];
+};
+
+const computePageTitleWithFullPath = (pageTitle, parents) => {
+    const maxAllowedPathSize =
+        MAX_TITLE_LENGTH - pageTitle.length - '...'.length;
+
+    pageTitle = pageTitle + computePath(parents, maxAllowedPathSize);
+
+    return pageTitle;
+};
+
+export const computePageTitle = (appName, study, parents) => {
+    if (!study) return appName;
+    let pageTitle = appName + SEPARATOR + study.elementName;
```

Comment:
You are dependent of the study Object format here. Seems a bad idea. Maybe this function should accept the studyName directly as string.

Reply:
like parents which is an array of parents names